docs/implementation.md

```diff
@@ -32,3 +32,14 @@ There are two primary entry points for middleware:
 
 1. `robot.listenerMiddleware` - registers a new piece of middleware in a global array
 2. `middleware.execute` - executes all registered middleware in order
+
+## Persistence
+
+Hubot has an in-memory key-value store exposed as `robot.brain` that can be used to store and retrieve data.
```

Comment:
"in-memory key-value" store may be misleading. When you say key-value, that typically suggets something like redis. In reality, it's just a javascript object, which technically has keys and values.